homeassistant/components/ecobee/climate.py

```diff
@@ -727,6 +755,63 @@ def resume_program(self, resume_all):
         )
         self.update_without_throttle = True
 
+    def set_sensors_used_in_climate(
+        self, sensors: list[str], climate_name: str | None = None
+    ) -> None:
+        """Set the sensors used on a climate for a thermostat."""
+        if climate_name is None:
+            climate_name = self.preset_mode
+
+        # Check if climate is an available preset option.
+        if climate_name not in self._preset_modes.values():
```

Comment:
```suggestion
        elif climate_name not in self._preset_modes.values():
```

Reply:
`elif` means else if
So the statement will only be evaluated if the previous if statement was false, which is un your case `if preset_mode is None:`

We set the preset_mode to the current one if it was None. As this is always a valid mode, we can skip this check.

Reply:
Gotcha. That makes sense.